tensorflow/core/kernels/cwise_op_leakyrelu_bf16.cc

```diff
@@ -0,0 +1,109 @@
+/* Copyright 2022 The TensorFlow Authors. All Rights Reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+==============================================================================*/
+
+#include "tensorflow/core/kernels/cwise_ops_common.h"
+#include "tensorflow/core/kernels/cwise_ops.h"
+#include "tensorflow/core/framework/op_kernel.h"
+
+#define _USE_MATH_DEFINES
+#include <cmath>
+#include <functional>
+#include <type_traits>
+
+#include "third_party/eigen3/unsupported/Eigen/CXX11/Tensor"
+#include "tensorflow/core/framework/bounds_check.h"
+#include "tensorflow/core/framework/numeric_types.h"
+#include "tensorflow/core/framework/tensor_types.h"
+
+// Keeping all new leakyrelu changes in 1 file.
+// This is similar to changes in cwise_ops.h
+namespace Eigen {
+namespace internal {
+
+template <typename Scalar>
+struct leakyrelu_op {
+  EIGEN_DEVICE_FUNC EIGEN_STRONG_INLINE
+  explicit leakyrelu_op(float val = 0.2f) EIGEN_NO_THROW {
+    alpha_ = Scalar(val);
+  }
+  EIGEN_DEVICE_FUNC EIGEN_STRONG_INLINE Scalar
+  operator()(const Scalar& x) const {
+    return x > Scalar(0) ? x : x * Scalar(alpha_);
+  }
+  template <typename Packet>
+  EIGEN_DEVICE_FUNC EIGEN_STRONG_INLINE Packet packetOp(const Packet& x) const {
+    Packet alpha = pset1<Packet>(alpha_);
+    return pselect(pcmp_le(x, pzero(x)), pmul(x,alpha), x);
```

Comment:
General note: `pcmp_le` and `pmul` here will end up performing the operation in `float`, which requires conversions back and forth.  Technically this will be converting `x` to `bfloat16` twice.  We don't (yet?) have a general approach for getting around this in Eigen yet, and it may not significantly impact performance at all - just thought I'd mention.